### PR TITLE
Add quest startup branch preparation

### PR DESCRIPTION
**Pull request by KjellKod**
## Summary
Add quest startup branch preparation so new quests can move off the default branch automatically while still supporting worktree and none modes.

## Changes
- add `scripts/quest_startup_branch.py` to create branch, worktree, or no-op startup context with a consistent blocked/created/skipped result contract
- wire startup branch metadata into `.skills/quest/SKILL.md` and `.skills/quest/delegation/workflow.md`
- add allowlist/schema, gitignore, manifest, checksum, and validation updates for the new helper
- add runtime tests for branch, worktree, none, dirty-tree, and invalid-allowlist cases

## Validation
- bash tests/test-quest-runtime.sh
- bash scripts/validate-quest-config.sh
- bash scripts/validate-manifest.sh

### CODE REVIEW

File: .ai/allowlist.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "plan_refinement": true,
     "implementation": false,
     "code_review": true,
-    "fix_loop": false
+    "fix_loop": true
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply .ai/allowlist.json:19]: Negative, we want this. It's just noise to be asked this when the answer is "yes" every time

File: scripts/quest_startup_branch.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+        return remote_head.rsplit("/", 1)[-1]
+
+    for candidate in ("main", "master"):
+        if git_success(repo_root, "show-ref", "--verify", "--quiet", f"refs/heads/{candidate}"):
+            return candidate
+
+    if current_branch in {"main", "master"}:
+        return current_branch
+
+    return "main"
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_startup_branch.py]: Fixed — removed the hardcoded `"main"` fallback. `detect_default_branch()` now returns `current_branch` when it can't determine the default from origin/HEAD or local main/master refs.

- Reviewed by Claude Opus 4.6, in collaboration with KjellKod

File: scripts/quest_startup_branch.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+            payload = build_result(
+                status="created",
+                branch=branch_name,
+                branch_mode="branch",
+                requested_branch_mode=requested_branch_mode,
+                current_branch=current_branch,
+                default_branch=default_branch,
+                branch_created=True,
+                worktree_path=None,
+                message=f"Created and checked out quest branch {branch_name}.",
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_startup_branch.py:285]: Fixed — added `SAFE_SLUG_RE` validation at the top of `main()`. Slugs must match `^[a-z0-9]([a-z0-9-]*[a-z0-9])?$` and cannot contain `..`, `/`, or `\\`. Invalid slugs return a `blocked` result before any path or branch operations.

- Reviewed by Claude Opus 4.6, in collaboration with KjellKod

File: .skills/quest/delegation/workflow.md [hunk 4]
```diff
@@ -258,7 +258,16 @@ This workflow expects to be invoked with a quest brief already prepared.
...
 2. If it does not exist, STOP and report error: "Quest brief not found. The routing layer should have created it before invoking workflow."
-3. If it exists, proceed to Step 2
+3. Read `.quest/<id>/state.json` and determine the source workspace root for code-bearing phases:
+   - If `worktree_path` exists and the directory is present, set `source_workspace_root = worktree_path`
+   - Otherwise set `source_workspace_root = <repo root>`
+   - All source edits plus `git status`, `git diff`, and `git log` commands in Steps 4-7 MUST run from `source_workspace_root`
+   - Quest artifacts always remain under `.quest/<id>/` in the original repo root; when `source_workspace_root != <repo root>`, prefer absolute quest artifact paths when invoking builder, reviewers, and fixer
+4. Verify branch context:
+   - If `branch` exists in state.json and `branch_mode == "branch"`, compare it to `git branch --show-current` in `source_workspace_root`
+   - If `branch_mode == "worktree"`, verify the directory at `worktree_path` still exists
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply .skills/quest/delegation/workflow.md:268]: Fixed — verification mismatch now stops and asks the user to confirm before continuing, rather than just warning. Running in the wrong workspace would defeat the startup isolation.

- Reviewed by Claude Opus 4.6, in collaboration with KjellKod

File: .ai/allowlist.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (KjellKod) [reply .ai/allowlist.json:19]: Intentional — maintainer confirmed this is wanted. The fix_loop gate was always answered 'yes', so flipping the default removes unnecessary friction. See KjellKod's reply above.

- Reviewed by Claude Opus 4.6, in collaboration with KjellKod

File: scripts/quest_startup_branch.py [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+        quest_link = worktree_path / ".quest"
+        quest_source = repo_root / ".quest"
+        if not quest_link.exists() and quest_source.exists():
+            quest_link.symlink_to(quest_source)
+
+        payload = build_result(
+            status="created",
+            branch=branch_name,
+            branch_mode="worktree",
+            requested_branch_mode=requested_branch_mode,
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_startup_branch.py:339]: Fixed — symlink is now created unconditionally (removed the `quest_source.exists()` guard). It may dangle briefly until quest folder creation runs in the next step, which is fine on macOS/Linux.

- Reviewed by Claude Opus 4.6, in collaboration with KjellKod

File: .quest-checksums [hunk 6]
```diff
@@ -52,8 +52,8 @@ c1ece37ca71c80ba1880631ac2762a79999b3812be243dd9a249a10065e3aab7  .skills/quest/
 e1cef5bb261421226a8007f88aab5cac0295b16f4044735856305c155a952376  .skills/quest/agents/README.md
 5e838b78b482e1c4933593dd57d7702ab03f5d3f2d10a062db5489ef184c9c88  .skills/quest/delegation/questioner.md
 c1138ae5c661af4333e465910dd864aa7d1c3a253f8696bfbcd2e679f5cf5801  .skills/quest/delegation/router.md
-5d32a6474a635fc0e9ba14dd92ca3852c0b6fd050494b624546d27c1675fedf0  .skills/quest/delegation/workflow.md
-75a7d855a0d5cad4f2ef0f0b7675ee6de550b8b53104e3968db7e66d477fef83  .skills/quest/SKILL.md
+a84c771b6837f2112652c4754aa410e647b1fcc4e33875c1a93539c06d2bd06a  .skills/quest/delegation/workflow.md
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply .quest-checksums]: Fixed — regenerated `.quest-checksums` from the current tracked Quest files and pushed `814254e`, so the installer integrity entries now match the shipped content on this PR head.

- Reviewed by GPT-5 Codex, in collaboration with KjellKod

File: .ai/allowlist.json [hunk 7]
```diff
@@ -42,12 +47,12 @@
     "plan_review_a": {
       "file_write": [".quest/**"],
       "file_read": ["**"],
-      "bash": ["git status", "git diff", "gh pr view.*"]
+      "bash": ["python", "python3", "pytest", "bash", "git status", "git diff", "git log", "rg", "grep", "find", "ls", "wc", "uv", "gh pr view.*"]
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply .ai/allowlist.json]: agreed, review models should not have access to bash, but the other ones are safe.

[the remark this answers is not in the document]
Comment (KjellKod) [reply .ai/allowlist.json]: Fixed — removed `bash` from `plan_review_a`, `plan_review_b`, and `code_review_agent` permission lists. Other read/analysis commands (python, pytest, git, rg, grep, find, ls, wc, uv, gh) are retained as the maintainer confirmed those are safe for reviewer roles.

- Reviewed by Claude Opus 4.6, in collaboration with KjellKod

File: .skills/quest/delegation/workflow.md [hunk 8]
```diff
@@ -258,7 +258,16 @@ This workflow expects to be invoked with a quest brief already prepared.
...
+   - If verification fails, STOP and ask the user to confirm before continuing — running in the wrong workspace defeats startup isolation
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply .skills/quest/delegation/workflow.md:270]: Fixed — worktree verification guidance now requires both the worktree path to exist and the checked-out branch in `source_workspace_root` to match `state.json.branch` when that branch is recorded, so a wrong-branch worktree is treated as a mismatch instead of silently continuing.

- Reviewed by GPT-5 Codex, in collaboration with KjellKod

File: ideas/renaming.md [hunk 9]
```diff
@@ -0,0 +1 @@
+Suggestion for quest: when installed, it adds several scripts files into the repo root's script/ folder, which is a fairly common tracked path across repositories and requires users to  add all of the added paths to .gitignore. I'd be nice if all the quest scripts would instead be instealled in scripts/quest/<here> or that they are all consistently prefixed with `quest_` like many of them already are
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply ideas/renaming.md]: Good catch. I removed `ideas/renaming.md` from this PR so the branch-preparation change stays focused and that idea can be handled separately.

- Reviewed by GPT-5 Codex, in collaboration with KjellKod

File: scripts/quest_startup_branch.py [hunk 10]
```diff
@@ -0,0 +1,355 @@
...
+    }
+
+
+def main() -> int:
+    args = parse_args()
+
+    if not SAFE_SLUG_RE.match(args.slug) or ".." in args.slug:
+        payload = build_result(
+            status="blocked",
+            branch=None,
```
[the remark this answers is not in the document]
Comment (KjellKod) [reply scripts/quest_startup_branch.py:148]: Fixed — invalid slug results now preserve the actual requested mode instead of hardcoding `branch`, and I added regression coverage in `tests/test-quest-runtime.sh` to pin that contract.

- Reviewed by GPT-5 Codex, in collaboration with KjellKod
